import_tracker/import_tracker.py

```diff
@@ -330,9 +332,39 @@ def _get_op_number(dis_line: str) -> Optional[int]:
     return int(line_parts[opcode_idx - 1])
 
 
-def _get_try_end_number(dis_line: str) -> int:
-    """For a SETUP_FINALLY/SETUP_EXPECT line, extract the target end line"""
-    return int(_get_value_col(dis_line).split()[-1])
+def _get_try_end_number(
+    dis_line: str,
+    op_num: Optional[int],
+    exception_table: Dict[int, int],
+) -> Optional[int]:
+    """If the line contains a known indicator for a try block, get the
+    corresponding end number
+
+    NOTE: This contains compatibility code for changes between 3.10 and 3.11
+    """
+    return exception_table.get(op_num or -1) or (
+        int(_get_value_col(dis_line).split()[-1])
+        if any(op in dis_line for op in ["SETUP_FINALLY", "SETUP_EXCEPT"])
+        else None
+    )
+
+
+def _get_exception_table(dis_lines: List[str]) -> Dict[int, int]:
+    """For 3.11+ exception handling, parse the Exception Table"""
+    table_start = [i for i, line in enumerate(dis_lines) if line == "ExceptionTable:"]
+    assert len(table_start) <= 1, "Found multiple exception tables!"
```

Comment:
I have not encountered a case where this is larger than 1 and can't imagine a circumstance under which it would be, but I haven't exhaustively read the spec either, so if this ever trips, we may need to investigate this further.